src/Microsoft.AspNetCore.Rewrite/ModRewrite/ModRewriteRule.cs

```diff
@@ -0,0 +1,203 @@
+            }
+
+            // 2. Go through all conditions and compare them to the created string
+            var previous = Match.Empty;
+
+            if (!CheckCondition(context, results, previous))
+            {
+                return new RuleResult { Result = RuleTerminiation.Continue };
+            }
+            // TODO add chained flag
+            
+            // at this point, our rule passed, we can now apply the on match function
+            var result = Transform.GetPattern(context.HttpContext, results, previous);
+
+            if (Flags.HasFlag(RuleFlagType.QSDiscard))
+            {
+                context.HttpContext.Request.QueryString = new QueryString();
+            }
+
+            if (Flags.HasFlag(RuleFlagType.Forbidden))
+            {
+                context.HttpContext.Response.StatusCode = 403;
+                return new RuleResult { Result = RuleTerminiation.ResponseComplete };
+            }
+            else if (Flags.HasFlag(RuleFlagType.Gone))
+            {
+                context.HttpContext.Response.StatusCode = 410;
+                return new RuleResult { Result = RuleTerminiation.ResponseComplete };
+            }
+            else if (result == "-")
+            {
+                // TODO set url to result.
+            }
+            else if (Flags.HasFlag(RuleFlagType.QSAppend))
+            {
+                context.HttpContext.Request.QueryString = context.HttpContext.Request.QueryString.Add(new QueryString(result));
+            } 
+            else
+            {
+                if (Flags.HasFlag(RuleFlagType.FullUrl))
+                {
+                    ModifyHttpContextFromUri(context.HttpContext, result);
+                }
+                else
+                {
+                    if (result.StartsWith("/"))
+                    {
+                        context.HttpContext.Request.Path = new PathString(ModRewriteBase.Base.Substring(1) + result);
+                    }
+                    else
+                    {
+                        context.HttpContext.Request.Path = new PathString(ModRewriteBase.Base + result);
+                    }
+                }
+            }
+
+            if ((flagRes = Flags.CheckFlag(RuleFlagType.Cookie)) != null)
+            {
+                // TODO CreateCookies(context);
+                // context.HttpContext.Response.Cookies.Append()
+                // Make sure this in on compile.
+            }
+
+            if (Flags.CheckFlag(RuleFlagType.Env) != null)
+            {
+                // TODO CreateEnv(context)
+                // context.HttpContext...
+            }
+
+            if (Flags.CheckFlag(RuleFlagType.Next) != null)
+            {
+
+            }
+
+            string statusCode;
+            if ((statusCode = Flags.CheckFlag(RuleFlagType.Redirect)) != null)
+            {
+                int parsedInt;
+                if (int.TryParse(statusCode, out parsedInt))
```

Comment:
Add a PERF TODO here: Parse the status code when the flag is parsed rather than per request.